extensions/resteasy-reactive/rest-client-microprofile/runtime/src/main/java/io/quarkus/resteasy/reactive/client/microprofile/MicroProfileRestClientResponseFilter.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.resteasy.reactive.client.microprofile;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientResponseContext;
+import javax.ws.rs.client.ClientResponseFilter;
+
+import org.eclipse.microprofile.rest.client.ext.ResponseExceptionMapper;
+import org.jboss.resteasy.reactive.client.handlers.ClientResponseRestHandler;
+import org.jboss.resteasy.reactive.client.impl.ClientRequestContextImpl;
+import org.jboss.resteasy.reactive.client.impl.ClientResponseContextImpl;
+import org.jboss.resteasy.reactive.common.jaxrs.ResponseImpl;
+
+public class MicroProfileRestClientResponseFilter implements ClientResponseFilter {
+    private final List<ResponseExceptionMapper<?>> exceptionMappers;
+
+    public MicroProfileRestClientResponseFilter(List<ResponseExceptionMapper<?>> exceptionMappers) {
+        if (exceptionMappers == null) {
+            throw new NullPointerException("exceptionMappers cannot be null");
+        }
+        this.exceptionMappers = exceptionMappers;
+    }
+
+    @Override
+    public void filter(ClientRequestContext requestContext, ClientResponseContext responseContext) throws IOException {
+        for (ResponseExceptionMapper exceptionMapper : exceptionMappers) {
```

Comment:
do we need to iterate this even for 2xx responses?

Reply:
[The spec](https://download.eclipse.org/microprofile/microprofile-rest-client-2.0/microprofile-rest-client-spec-2.0.html#_how_to_implement_responseexceptionmapper) says 
```
Likewise, the handles method by default will handle any response status code >= 400. You may override this behavior if you so choose to handle other response codes (both a smaller ranger and a larger range are expected) or base the decision on the response headers.
```

My understanding of ^ is that we need.